packages/edit-site/src/components/sidebar-navigation-screen-patterns/category-item.js

```diff
@@ -12,19 +12,11 @@ export default function CategoryItem( {
 	label,
 	type,
 } ) {
-	const linkInfo = useLink(
-		{
-			path: '/patterns',
-			categoryType: type,
-			categoryId: id,
-		},
-		{
-			// Keep a record of where we came from in state so we can
-			// use the browser's back button to go back to Patterns.
-			// See the implementation of the back button in patterns-list.
-			backPath: '/patterns',
```

Comment:
Unfortunately, this makes mobile back button **push** to the history stack instead of backing when available.

I believe this issue was introduced in https://github.com/WordPress/gutenberg/pull/52910 but I haven't looked deep into the issue to provide any valuable insight 😅 . (c.c. @noisysocks)

Reply:
I tested `trunk` and looks like it's working as expected. The back button goes "up" not "back" and so therefore creates a new entry in the browser's history. That's how the other screens work too e.g. if you go to Pages → About, click the back button, and then go back with your browser, you end up at About.

Reply:
What I mean is I expect the back button to behave exactly like the browsers' back button whenever possible. Something like a [smart back button](https://developer.mozilla.org/en-US/docs/Web/API/Navigation/entries#a_smart_back_button).

I don't think this has that high of a priority though 🙂 . Perhaps we could revisit this once we have a more robust routing architecture.